### PR TITLE
Add a Jenkinsfile to run locally

### PR DESCRIPTION
**Pull request by Arsnael**
Almost there...

At the end I checked the code in jenkinsfile-runner and tweaked their `adoptopenjdk-11-hotspot` dockerfile to have something to run our build, by adding git, maven 3.6.3 and docker: https://github.com/Arsnael/jenkinsfile-runner/commit/ac9687cbde1a862da0b75049207a38aadc1c36d6

Then I built it locally and pushed it on my docker hub account => `rcordier/jenkinsfile-runner:maven-jdk11`

To run this Jenkinsfile locally, from the root of this repo type:

```
docker run --rm -v /var/run/docker.sock:/var/run/docker.sock -v $HOME/.m2:/root/.m2 -v $(pwd)/Jenkinsfile:/workspace/Jenkinsfile rcordier/jenkinsfile-runner:maven-jdk11
```

The `-v $HOME/.m2:/root/.m2` is because I don't like to wait for all artifacts being dl from maven every time I run it :D 

It's building. The tests are almost all passing except the JMAP deployment tests (that's the only one failing... for all implems):

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 13.599 s <<< FAILURE! - in com.linagora.openpaas.deployment.MemoryJmapTest
[ERROR] mailboxGetTest  Time elapsed: 13.597 s  <<< ERROR!
java.net.ConnectException: Connection refused (Connection refused)
```

Not sure yet what is the issue... Maybe I need to expose the port 80? Will debug that tomorrow

### CODE REVIEW

**Review by chibenwa — COMMENTED** [review]
Please add in the README how to use the jenkins runner.

Please include the DockerFile of the runner here too so that people can also build it instead of downloading it from your dockerHub repos.

**Comment by Arsnael** [comment]
> Please include the DockerFile of the runner here too so that people can also build it instead of downloading it from your dockerHub repos.

That's a bit more complicated... It takes and copies a bunch of files in that repo... Not sure we want to copy the all thing.

Also I thought this should be temporary until the CI is up and running? I'm really starting to wonder if that task makes sense, as the Jenkinsfile is pretty basic and the main effort is spent on a standalone runner, where the conf for Jenkins2 will probably be different and require a different work

**Comment by chibenwa** [comment]
> That's a bit more complicated... It takes and copies a bunch of files in that repo... Not sure we want to copy the all thing.

Can't we use the original runner in the `FROM` ?

**Comment by Arsnael** [comment]
> Can't we use the original runner in the `FROM` ?

Could try some dark magic around it I guess

**Comment by Arsnael** [comment]
Ok now the deployment tests are finally passing. I will write the doc and try to add some jenkins-runner dockerfile for allowing people to build it locally

**Comment by Arsnael** [comment]
Ready for review. Running the full build locally (but with the m2 local repository mounted to not redl everything), it took me 20 minutes in total.

File: README.md [hunk 1]
```diff
@@ -44,4 +46,31 @@ Then you can compile both `apache/james-project` and `linagora/openpaas-james` t
...
+
+* checkout and compile latest code of Apache James project alongside `openpaas-james`
+* generate docker images for `memory`, `distributed` and `distributed-ldap` flavors of the project
+* launch unit, integration and deployment tests on `openpaas-james`
+
+To launch it you need to have docker installed. From the root of this project, you can either use directly the image
+provided on docker hub:
+
+```
+docker run --rm -v /var/run/docker.sock:/var/run/docker.sock -v $(pwd)/Jenkinsfile:/workspace/Jenkinsfile rcordier/jenkinsfile-runner:maven-jdk11
```
Comment (chibenwa) [inline README.md:64]: let's not advertize this option and only ask people to build the jenkins-runner themselves

**Review by chibenwa — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Arsnael) [reply README.md]: Done :)

**Review by chibenwa — APPROVED** [review]

**Review by LanKhuat — APPROVED** [review]

**Pull request merged**